cyences_app_for_splunk/bin/cyences_email_utility.py

```diff
@@ -263,12 +233,41 @@ def _get_smtp_creds_util(self):
             self.logger.error("Could not get email credentials from splunk, using no credentials. Error: %s" % (str(e)))
 
         return '', ''
+    
+
+    def filter_recipients_if_not_in_allowed_domains(self, recipients):
+        if self.emailConfigs['allowedDomainList'] != "" and self.emailConfigs['allowedDomainList'] != None:
+            validRecipients = []
+
+            domains = []
+            domains.extend(EMAIL_DELIM.split(self.emailConfigs['allowedDomainList']))
+            domains = [d.strip() for d in domains]
+            domains = [d.lower() for d in domains]
+
+            recipients = [r.lower() for r in recipients]
+
+            for recipient in recipients:
+                dom = recipient.partition("@")[2]
+                if not dom in domains:
+                    self.logger.error("Email recipient=%s is not among the allowedDomainList=%s in alert_actions.conf file. Removing it from the recipients list."
+                                % (recipient, self.emailConfigs['allowedDomainList']))
+                else:
+                    validRecipients.append(recipient)
+
+            if len(validRecipients) != len(recipients):
+                self.logger.error("Not all of the recipient email domains are on the allowed domain list. Sending emails only to %s" % str(validRecipients))
+            
+            return validRecipients
+
+        else:
+            return recipients
 
 
-    def send(self, to, cc='', bcc='', subject='Splunk Alert', html_body='', results_link=''):
+    def send(self, to, cc=[], bcc=[], subject='Splunk Alert', html_body=''):
```

Comment:
If we use object as a default argument (list, or dict), it will keep reference of previously called item instead of passing the empty list.